### PR TITLE
[DEV][BACKEND] Comment Backend

### PR DESCRIPTION
**Pull request by Oguzoz1**
-> Create Comment
-> Delete Comment

Backend service has been added. 

Newly formed classes:
-> Comment Controller: RestController
-> CommentDto: Transfer Object
-> Comment: Entity
-> Comment Repository: Repository
-> Comment Service: Service for dealing comment

Edited:
-> Post: Entity

### CODE REVIEW

**Comment by niloaydin** [comment]
@Oguzoz1 Can you please change the title for this merge request according to the PR standards. You can find the standards through [this link](https://github.com/SWE574-G3/SWE-574-G3/wiki/Coding-Standards#pull-request-guidelines).

File: Communitter/src/main/java/com/communitter/api/service/CommentService.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+@Service
+@RequiredArgsConstructor
+public class CommentService {
+    private final CommentRepository commentRepository;
+    private final PostRepository postRepository;
+    public Logger logger = LoggerFactory.getLogger(JwtService.class);
+
+    @Transactional
+    public CommentDto createComment(Long postId, CommentDto commentDto){
+        Post post = postRepository.findById(postId).orElseThrow();
```
Comment (niloaydin) [inline Communitter/src/main/java/com/communitter/api/service/CommentService.java:26]: can we throw specific errors here

File: Communitter/src/main/java/com/communitter/api/service/CommentService.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        logger.info("Date: {}", createdComment.getDate());
+        logger.info("Post ID: {}", createdComment.getPost().getId());
+        logger.info("Author ID: {}", createdComment.getAuthor().getId());
+        commentRepository.save(createdComment);
+        return commentDto;
+    }
+
+    @Transactional
+    public String deleteComment(Long id){
+        Comment comment = commentRepository.findById(id).orElseThrow();
```
Comment (niloaydin) [inline Communitter/src/main/java/com/communitter/api/service/CommentService.java:39]: comments can be deleted either by the person who created it or the owner. Where do you do this check?

File: Communitter/src/main/java/com/communitter/api/service/CommentService.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class CommentService {
+    private final CommentRepository commentRepository;
+    private final PostRepository postRepository;
+    public Logger logger = LoggerFactory.getLogger(JwtService.class);
+
+    @Transactional
+    public CommentDto createComment(Long postId, CommentDto commentDto){
```
Comment (niloaydin) [inline Communitter/src/main/java/com/communitter/api/service/CommentService.java:25]: can we add try catch block, so we can throw our exceptions we created according to the situation

File: Communitter/src/main/java/com/communitter/api/service/CommentService.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+
+@Service
+@RequiredArgsConstructor
+public class CommentService {
+    private final CommentRepository commentRepository;
+    private final PostRepository postRepository;
+    public Logger logger = LoggerFactory.getLogger(JwtService.class);
```
Comment (niloaydin) [inline Communitter/src/main/java/com/communitter/api/service/CommentService.java]: can we change the service in here, it should be commentservice

File: Communitter/src/main/java/com/communitter/api/service/CommentService.java (hunk 2 above)
Comment (niloaydin) [reply Communitter/src/main/java/com/communitter/api/service/CommentService.java:39]: Also can we add try catch here as well

File: Communitter/src/main/java/com/communitter/api/service/CommentService.java (hunk 4 above)
Comment (Oguzoz1) [reply Communitter/src/main/java/com/communitter/api/service/CommentService.java]: what do you mean

Comment (niloaydin) [reply Communitter/src/main/java/com/communitter/api/service/CommentService.java]: @Oguzoz1 logger icin service yaziyoruz ya icine. jwt service degil de comment servicete kullaniyoruz comment service olcak

Comment (Oguzoz1) [reply Communitter/src/main/java/com/communitter/api/service/CommentService.java]: Komple siliyorum

File: Communitter/src/main/java/com/communitter/api/service/CommentService.java (hunk 2 above)
Comment (Oguzoz1) [reply Communitter/src/main/java/com/communitter/api/service/CommentService.java:39]: > comments can be deleted either by the person who created it or the owner. Where do you do this check?

image

Comment (Oguzoz1) [reply Communitter/src/main/java/com/communitter/api/service/CommentService.java:39]: I will be adding those backend checkers once I engage editing tasks

**Review by niloaydin — APPROVED** [review]

**Pull request merged**